photon-ml/src/main/scala/com/linkedin/photon/ml/Params.scala

```diff
@@ -158,6 +160,9 @@ class Params {
    */
   var deleteOutputDirsIfExist: Boolean = false
 
+  var inputFormatType: InputFormatType = InputFormatType.AVRO
+  var featureDimension: Int = -1
```

Comment:
It's one part of the reason that we need to indicate feature dimension when constructing `SparseVector`. But it's also because presumably LibSVM format is already a pre-indexed format that I feel it's a bit redundant to do second round indexing just to learn the feature dimension. (Although it might make some sense in some cases when the original indexing is not dense enough). So I'm inclined to keep it simple for now but later add in other options more properly if needed. But we do need to be specific that `featureDimension` is a parameter only used by LibSVM format. What do you think?